Userland/Libraries/LibCrypto/BigFraction/BigFraction.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2022, Lucas Chollet <lucas.chollet@free.fr>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "BigFraction.h"
+#include <AK/Math.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+#include <LibCrypto/NumberTheory/ModularFunctions.h>
+
+namespace Crypto {
+
+BigFraction::BigFraction(Crypto::SignedBigInteger numerator, Crypto::UnsignedBigInteger denominator)
+    : m_numerator(move(numerator))
+    , m_denominator(move(denominator))
+{
+    VERIFY(m_denominator != 0);
+    reduce();
+}
+
+BigFraction::BigFraction(Crypto::SignedBigInteger value)
+    : BigFraction(move(value), 1)
+{
+}
+
+BigFraction::BigFraction(StringView sv)
```

Comment:
FIXME?

Reply:
Isn't the one in BigFraction.h enough?

Reply:
I'd say no because that's moderately far away from this, and "propagate errors" is too general.